apitools/base/py/credentials_lib.py

```diff
@@ -129,6 +134,28 @@ def ServiceAccountCredentials(service_account_name, private_key, scopes,
         service_account_name, private_key, scopes, **service_account_kwargs)
 
 
+def _GetApplicationDefaultCredentials(scopes):
+    gc = oauth2client.client.GoogleCredentials
+    with cache_file_lock:
+        try:
+            # pylint: disable=protected-access
+            # We've already done our own check for GAE/GCE
+            # credentials, we don't want to pay for checking again.
+            credentials = gc._implicit_credentials_from_files()
+        except oauth2client.client.ApplicationDefaultCredentialsError:
+            return None
+    # If we got back a non-service-account credential, we need to use
+    # a heuristic to decide whether or not the application default
+    # credential will work for us. We assume that if we're requesting
+    # cloud-platform, our scopes are a subset of cloud scopes, and the
+    # ADC will work.
+    if not isinstance(credentials, gc):
```

Comment:
yep -- but there was no way to format it so that the linter didn't complain. :wink: 

fixing, because i don't let no linter boss ME around.